x-pack/plugins/ml/public/embeddables/single_metric_viewer/single_metric_viewer_embeddable_factory.tsx

```diff
@@ -0,0 +1,331 @@
+
+          // Holds the container height for previously fetched data
+          const containerHeightRef = useRef<number>();
+
+          useEffect(function setUpJobsLoaded() {
+            async function loadJobs() {
+              try {
+                await mlJobService.loadJobsWrapper();
+                setJobsLoaded(true);
+              } catch (e) {
+                const errorMessage = extractErrorMessage(e);
+                setError(errorMessage);
+              }
+            }
+            loadJobs();
+            // eslint-disable-next-line react-hooks/exhaustive-deps
+          }, []);
+
+          useEffect(
+            function setUpSelectedJob() {
+              async function fetchSelectedJob() {
+                if (mlApiServices && selectedJobId !== undefined && error === undefined) {
+                  try {
+                    const { jobs } = await mlApiServices.getJobs({ jobId: selectedJobId });
+                    const job = jobs[0];
+                    setSelectedJob(job);
+                  } catch (e) {
+                    const errorMessage = extractErrorMessage(e);
+                    setError(errorMessage);
+                  }
+                }
+              }
+              fetchSelectedJob();
+            },
+            [selectedJobId, mlApiServices, error]
+          );
+
+          const autoZoomDuration = useMemo(() => {
+            if (!selectedJob) return;
+            return mlTimeSeriesExplorerService?.getAutoZoomDuration(selectedJob);
+          }, [mlTimeSeriesExplorerService, selectedJob]);
+
+          // eslint-disable-next-line react-hooks/exhaustive-deps
+          const resizeHandler = useCallback(
+            throttle((e: { width: number; height: number }) => {
+              // Keep previous container height so it doesn't change the page layout
+              containerHeightRef.current = e.height;
+
+              if (Math.abs(chartWidth - e.width) > minElemAndChartDiff) {
+                setChartWidth(e.width);
+              }
+            }, RESIZE_THROTTLE_TIME_MS),
+            [chartWidth]
+          );
+
+          const appStateHandler = useCallback(
+            (action: string, payload?: any) => {
+              /**
+               * Empty zoom indicates that chart hasn't been rendered yet,
+               * hence any updates prior that should replace the URL state.
+               */
+              switch (action) {
+                case APP_STATE_ACTION.SET_FORECAST_ID:
+                  setSelectedForecastId(payload);
+                  setZoom(undefined);
+                  break;
+
+                case APP_STATE_ACTION.SET_ZOOM:
+                  setZoom(payload);
+                  break;
+
+                case APP_STATE_ACTION.UNSET_ZOOM:
+                  setZoom(undefined);
+                  break;
+              }
+            },
+
+            [setZoom, setSelectedForecastId]
+          );
+
+          if (error) {
```

Comment:
Instead of writing your own error display like this, it would be a lot easier to use the `publishesBlockingError` interface defined in `packages/presentation/presentation_publishing/interfaces/publishes_blocking_error.ts`, that way your error is integrated in the ecosystem. i.e. you get the error rendering / recovery for free, and any actions which shouldn't be available on panels with errors will not be.

Reply:
Good call - updated in 80ec80413f27de6f6cd4ba37746a9dc55bcc8038

Reply:
I see you've started using `blockingError`, but you're still rendering your own error. I think if you call `blockingError.next()` the error display you've built won't even be shown.